docs/design/design.md

```diff
@@ -0,0 +1,22 @@
+# Design Documentation
+
+This is an attempt to document the OpenShift Installer's existing design principles and also articulate design goals to guide with new feature development and in some cases inform bug fixes.
+
+## Design Goals
+
+### Install methods
+The OpenShift Installer currently supports 2 installation types, UPI (User Provisioned Infrastructure) and IPI (Installer Provisionined Infrastructure). IPI is a highly opinionated method of installing OpenShift. The IPI installation method tries to automate a large portion of the OpenShift Install with the least amount of intervention from the user. While it tries to take care of supporting the most common install scenarios, it cannot be expected to support every install variation that a user wants to use for their cloud.
+
+This is where the UPI offers the greater flexibility required for these highly specialized installs. In this install method, the user is expected to provide and manage several non-OpenShift components needed for the install on their own. For this reason, documentation surrounding the UPI flow are especially important.
```

Comment:
> the user is expected to provide and manage several non-OpenShift components needed for the install on their own

I would nitpick on the words "manage" and "several non-OpenShift components." Manage sounds like an ongoing process, whereas day-2 UPI & IPI should be indistinguishable. 

"Infrastructure" would be a bit more precise, IMHO, than non-OpenShift components.